package/MDAnalysis/analysis/base.py

```diff
@@ -85,8 +86,8 @@ class in `scikit-learn`_.
     def _validate_key(self, key):
         if key in dir(dict):
             raise TypeError(f"'{key}' is a protected dictionary attribute")
-        elif not (isinstance(key, str) and key[0].isalpha()
-                  and re.match("^[a-zA-Z0-9_]*$", key)):
+        elif not (isinstance(key, str)
+                  and re.match("^[a-zA-Z_][a-zA-Z0-9_]*$", key)):
             raise TypeError(f"'{key}' is not able to be accessed by attribute")
```

Comment:
Is `TypeError` the appropriate exception here or should it be `ValueError` (because you provided an incorrect value)?